src/github_repo_request_local.py

```diff
@@ -0,0 +1,74 @@
+import subprocess
+from pathlib import Path
+import pandas as pd
+from loguru import logger
+import shutil
+
+
+def count_and_list_test_files(directory):
+    logger.info(f"Processing the directory {directory}")
+    test_files = []
+    # Define a set of file extensions to exclude
+    excluded_extensions = {
+        ".txt",
+        ".md",
+        ".h",
+        ".xml",
+        ".html",
+        ".json",
+        ".png",
+        ".jpg",
+    }
+    for path in directory.rglob("*"):
```

Comment:
Are these paths that are returned by `directory.rglob(...)`? It might be clearer to use the term `item` as both filenames and paths can be returned. See https://realpython.com/get-all-files-in-directory-python/#:~:text=To%20recursively%20list%20the%20items,their%20subdirectories%2C%20and%20so%20on.&text=.py')%5D-,The%20.,from%20the%20Path%20object%20recursively.

Reply:
- It returns all the files and folders from the Path object recursively.
- Change the loop variable in the function count_and_list_test_files from `path` to `item` for accuracy
- changes have been committed